src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/BuildHistoryManager.java

```diff
@@ -40,23 +40,26 @@ public BuildHistoryManager(List<Rule> rules) {
      */
     @Override
     public synchronized void perform(Job<?, ?> job) throws IOException, InterruptedException {
+        String uniquePerformName = job.getFullName();
+        LOG.info(uniquePerformName + ": start evaluating build history");
+
         // reset counters of matched builds
         for (Rule rule : rules) {
-            rule.initialize();
+            rule.initialize(uniquePerformName);
         }
 
         Run<?, ?> run = job.getLastCompletedBuild();
         // for each completed build...
         while (run != null) {
-            LOG.info("Processing build #" + run.getNumber());
+            LOG.info(uniquePerformName + ": Processing build #" + run.getNumber());
```

Comment:
since this method is synchronized I don't think there is value to pass this into each log

Reply:
synchronized only on instance execution isn't it?
so there is the possibility that two different projects are analyzed in parallel

Reply:
but then you have two separate logs - is that right?

I'm not good at case which you currently having so that's why I'm asking

Reply:
no this is not the console output of the current build, this is part of the system log.
if you want to log to the console output you need a RunListener and call getLogger() on that

Reply:
![image](https://user-images.githubusercontent.com/14205420/106386103-ac9aaf80-63d3-11eb-9172-6d1e2346512a.png)

this is what the log looks like with the changes and 2 projects only